controller/transfer/project_transfer_manager.py

```diff
@@ -287,11 +288,169 @@ def import_file(
+            else:
+                attribute_name = __get_attribute_name_from_embedding_name(
+                    embedding_name
+                )
+                attribute_id = attribute_ids_by_old_name[attribute_name]
+
+            finished_at_str = "finished_at" in embedding_item
+            if not finished_at_str:
+                embedding_item["finished_at"] = sql.func.now()
+
+            embedding_object = embedding.create(
+                project_id=project_id,
+                attribute_id=attribute_id,
+                name=embedding_name,
+                state="FINISHED",
+                created_by=import_user_id,
+                custom=embedding_item.get(
+                    "custom",
+                ),
+                type=embedding_item.get(
+                    "type",
+                ),
+                started_at=embedding_item.get(
+                    "started_at",
+                ),
+                finished_at=embedding_item.get(
+                    "finished_at",
+                ),
+                platform=embedding_item.get(
+                    "platform",
+                ),
+                model=embedding_item.get(
+                    "model",
+                ),    
+            )
+            embedding_ids[
+                embedding_item.get(
+                    "id",
+                )
+            ] = embedding_object.id
+
+        if data.get(
+            "embedding_tensors_data",
+        ): 
+            for embedding_tensor_item in data.get(
+                "embedding_tensors_data",
+            ):
+                embedding.create_tensor(
+                    project_id=project_id,
+                    record_id=record_ids.get(
+                        embedding_tensor_item.get(
+                            "record_id",
+                        )
+                    ),
+                    embedding_id=embedding_ids.get(
+                        embedding_tensor_item.get(
+                            "embedding_id",
+                        )
+                    ),
+                    data=embedding_tensor_item.get(
+                        "data",
+                    ),
+                )
+
+    def __replace_embedding_name(soure_code: str, embedding_name_mapping: Dict[str, str]) -> str:
+        for embedding_name in embedding_name_mapping.keys():
+            double_quoted_name = f'"{embedding_name}"'
+            single_quoted_name = f"'{embedding_name}'"
+            if double_quoted_name in soure_code:
+                code =  soure_code.replace(
+                    double_quoted_name,
+                    f'"{embedding_name_mapping[embedding_name]}"')
+                return code
+            elif single_quoted_name in soure_code:
+                code =  soure_code.replace(
+                    single_quoted_name,
+                    f"'{embedding_name_mapping[embedding_name]}'")
+                return code
+        return soure_code
```

Comment:
this could result in an issue if both exist or more than one embedding name is present (not sure how likely but since customer code at least possible).
So maybe without return and elif = if
```suggestion
    def __replace_embedding_name(soure_code: str, embedding_name_mapping: Dict[str, str]) -> str:
        code = source_code
        for embedding_name in embedding_name_mapping.keys():
            double_quoted_name = f'"{embedding_name}"'
            single_quoted_name = f"'{embedding_name}'"
            if double_quoted_name in code:
                code =  code.replace(
                    double_quoted_name,
                    f'"{embedding_name_mapping[embedding_name]}"')
            if single_quoted_name in soure_code:
                code =  code.replace(
                    single_quoted_name,
                    f"'{embedding_name_mapping[embedding_name]}'")
        return code 
```